target/linux/realtek/files-6.12/drivers/net/ethernet/rtl838x_eth.h

```diff
@@ -309,7 +309,10 @@ inline u32 rtl930x_get_mac_link_sts(int port)
 
 inline u32 rtl931x_get_mac_link_sts(int p)
 {
-	return (sw_r32(RTL931X_MAC_LINK_STS + ((p >> 5) << 2)) & BIT(p % 32));
+	u32 link = sw_r32(RTL931X_MAC_LINK_STS + ((p >> 5) << 2));
+	link = sw_r32(RTL931X_MAC_LINK_STS + ((p >> 5) << 2));
+
+	return (link & BIT(p % 32));
```

Comment:
Looks strange. For better readability please write the whole function like this and rename it to the new naming scheme.

```
inline u32 rtldsa_931x_get_mac_link_sts(int port)
{
  int link, mask = BIT(port % 32), reg = RTL931X_MAC_LINK_STS + (port / 32) * 4;

  link = sw_r32(reg);
  link = sw_r32(reg);

  return (link & mask);
}
```
Wen this and the other changes are applied. This is ready for approval.